nvbench/detail/statistics.cuh

```diff
@@ -56,8 +58,124 @@ ValueType standard_deviation(Iter first, Iter last, ValueType mean)
                                                             val *= val;
                                                             return val;
                                                           }) /
-                        static_cast<ValueType>((num - 1));
+                        static_cast<ValueType>((num - 1)); // Bessel’s correction
   return std::sqrt(variance);
 }
 
+template <class It>
+nvbench::float64_t compute_mean(It first, It last)
+{
+  const auto n = std::distance(first, last);
+  return std::accumulate(first, last, 0.0) / static_cast<nvbench::float64_t>(n);
```

Comment:
This has a divide by 0 error if the iterator has a size 0. Previously this was guarded by the size check at line 84, but now this division is done before that is checked.